test/cli/whole-program_test.py

```diff
@@ -214,6 +214,21 @@ def test_suppress_inline_project_builddir_j(tmpdir):
     os.mkdir(build_dir)
     __test_suppress_inline_project(tmpdir, ['-j2', '--cppcheck-build-dir={}'.format(build_dir)])
 
+@pytest.mark.parametrize("jobs", [1, 2])
+def test_addon_builddir_rerun(tmpdir, jobs):
+    """Rerun analysis with a build dir and ensure that misra CTU works"""
+    args = [
+        f'--cppcheck-build-dir={tmpdir}',
+        f'-j{jobs}',
+        '--addon=misra',
+        '--enable=style',
+        '--template={id}',
+        'whole-program']
+    _, _, stderr = cppcheck(args, cwd=__script_dir)
+    assert 'misra-c2012-5.8' in stderr
+    _, _, stderr = cppcheck(args, cwd=__script_dir)
+    assert 'misra-c2012-5.8' in stderr
```

Comment:
My intention was only to refactor. But here is a test case that does not work in cppcheck main branch but does work in this branch.